### PR TITLE
Fix more than half of the trash

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: cmake/warnings.cmake [hunk 1]
```diff
@@ -24,7 +24,10 @@ if (COMPILER_CLANG)
     no_warning(c++98-compat-pedantic)
     no_warning(c++98-compat)
     no_warning(c++20-compat) # Use constinit in C++20 without warnings
-    no_warning(conversion)
```
Comment (alexey-milovidov) [inline cmake/warnings.cmake:27]: Here it is.

Comment (alexey-milovidov) [reply cmake/warnings.cmake:27]: This is needed for #41168

File: src/Storages/examples/merge_selector2.cpp [hunk 2]
```diff
@@ -115,9 +115,9 @@ int main(int, char **)
 
         double time_to_merge = sum_merged_size / (1048576 * 10.0);
 
-        age_passed += time_to_merge;
+        age_passed = static_cast<size_t>(age_passed + time_to_merge);
         for (auto & part : parts)
-            part.age += time_to_merge;
+            part.age = static_cast<size_t>(part.age + time_to_merge);
```
Comment (rschu1ze) [inline src/Storages/examples/merge_selector2.cpp:120]:
```suggestion
            part.age = static_cast<time_t>(part.age + time_to_merge);
```

File: src/Storages/MergeTree/MergeTreePartsMover.cpp [hunk 3]
```diff
@@ -111,7 +111,7 @@ bool MergeTreePartsMover::selectPartsForMove(
         {
             for (const auto & disk : volumes[i]->getDisks())
             {
-                UInt64 required_maximum_available_space = disk->getTotalSpace() * policy->getMoveFactor();
+                UInt64 required_maximum_available_space = static_cast<size_t>(disk->getTotalSpace() * policy->getMoveFactor());
```
Comment (rschu1ze) [inline src/Storages/MergeTree/MergeTreePartsMover.cpp:114]:
```suggestion
                UInt64 required_maximum_available_space = static_cast<UInt64>(disk->getTotalSpace() * policy->getMoveFactor());
```

File: src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp [hunk 4]
```diff
@@ -98,10 +98,10 @@ UInt64 MergeTreeDataMergerMutator::getMaxSourcePartsSizeForMerge(size_t max_coun
     if (scheduled_tasks_count <= 1 || free_entries >= data_settings->number_of_free_entries_in_pool_to_lower_max_size_of_merge)
         max_size = data_settings->max_bytes_to_merge_at_max_space_in_pool;
     else
-        max_size = interpolateExponential(
+        max_size = static_cast<size_t>(interpolateExponential(
```
Comment (rschu1ze) [inline src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp:101]:
```suggestion
        max_size = static_cast<UInt64>(interpolateExponential(
```

File: src/Functions/greatCircleDistance.cpp [hunk 5]
```diff
@@ -44,17 +44,19 @@ namespace
 {
 
 constexpr double PI = 3.14159265358979323846;
+constexpr float PI_F = 3.14159265358979323846f;
```
Comment (rschu1ze) [inline src/Functions/greatCircleDistance.cpp:47]: (C++20:

```cpp
#include <numbers>

constexpr double PI = std::numbers::pi_v<double>;
constexpr float PI_F = std::numbers::pi_v<float>;
```
)

File: src/Functions/sleep.h [hunk 6]
```diff
@@ -100,7 +100,7 @@ class FunctionSleep : public IFunction
                 throw Exception("The maximum sleep time is 3 seconds. Requested: " + toString(seconds), ErrorCodes::TOO_SLOW);
 
             UInt64 count = (variant == FunctionSleepVariant::PerBlock ? 1 : size);
-            UInt64 microseconds = seconds * count * 1e6;
+            UInt64 microseconds = static_cast<size_t>(seconds * count * 1e6);
```
Comment (rschu1ze) [inline src/Functions/sleep.h:103]:
```suggestion
            UInt64 microseconds = static_cast<UInt64>(seconds * count * 1e6);
```

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**